resources/lib/stillwatching.py

```diff
@@ -57,9 +57,11 @@ def set_info(self):
             self.setProperty('playcount', str(self.item.get('playcount', 0)))
 
     def prepare_progress_control(self):
-        self.progress_control = self.getControl(3014)
-        if self.progress_control is not None:
+        try:
+            self.progress_control = self.getControl(3014)
             self.progress_control.setPercent(self.current_progress_percent)  # pylint: disable=no-member
+        except RuntimeError:
```

Comment:
Only use a try-block for the statement you know will throw the exception, if you still need to perform tasks when there is no exception, please use an else-block.

It is also a good idea to document when this specific exception is thrown, so people after you know the exact case you are covering.

So:
```python
        try:
            self.progress_control = self.getControl(3014)
        except RuntimeError:  # Happens when the skin has no progress bar control
            pass
        else:
            self.progress_control.setPercent(self.current_progress_percent)  # pylint: disable=no-member	            
```

Reply:
Please fix this for the other occurrences as well. Thanks !

Reply:
Fair point. Will fix this.